# 360° Digital Marketing Performance Audit
#### **Context**
This analysis is based on simulated digital marketing campaign data for three
major Indian beauty e-commerce brands : Nykaa, Purplle and Tira Beauty.
The dataset covers multiple channels, campaign types, customer segments and languages.

#### **Business Problem**
How are Nykaa, Purplle and Tira Beauty performing across their digital marketing
campaigns, and where does each brand have optimization opportunities ?

#### **Objectives**
The goal of this analysis is to :
1. Evaluate acquisition & reach efficiency by channel and brand
2. Assess conversion and revenue performance by campaign type and customer segment
3. Benchmark cost efficiency across brands and channels
4. Understand audience engagement patterns by segment and language
5. Analyze time performance and detect seasonality or campaign peaks

#### **KPIs of Focus**
- CTR (Click-Through Rate) = Clicks / Impressions
- CVR (Conversion Rate) = Conversions / Clicks
- CPL (Cost per Lead) = Acquisition_Cost / Leads
- ROI, Revenue, Engagement_Score

#### **1- Setup & Data Loading**
This section covers library imports, data loading, individual file verification,
merging and data type fixing.

In [ ]:
# Importing libraries

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
# Loading each brand's campaign data from separate CSV files

df_nykaa = pd.read_csv("/content/drive/MyDrive/Data projects/Multi-Brand Marketing Campaign/nykaa_campaign_data.csv")
df_purplle = pd.read_csv("/content/drive/MyDrive/Data projects/Multi-Brand Marketing Campaign/purplle_campaign_data.csv")
df_tira = pd.read_csv("/content/drive/MyDrive/Data projects/Multi-Brand Marketing Campaign/tira_campaign_data.csv")

# Quick verification function
def check_df(df, name):
    print(f"{'-'*40}")
    print(f"Brand : {name}")
    print(f"Shape : {df.shape}")
    print(f"Columns : {df.columns.tolist()}")
    print(f"Missing values : {df.isnull().sum().sum()}")
    print(f"Duplicates : {df.duplicated().sum()}")
    print(f"\nFirst row :")
    print(df.head(1))
    print()

check_df(df_nykaa, "Nykaa")
check_df(df_purplle, "Purplle")
check_df(df_tira, "Tira Beauty")



----------------------------------------
Brand : Nykaa
Shape : (55555, 16)
Columns : ['Campaign_ID', 'Campaign_Type', 'Target_Audience', 'Duration', 'Channel_Used', 'Impressions', 'Clicks', 'Leads', 'Conversions', 'Revenue', 'Acquisition_Cost', 'ROI', 'Language', 'Engagement_Score', 'Customer_Segment', 'Date']
Missing values : 0
Duplicates : 0

First row :
   Campaign_ID Campaign_Type   Target_Audience  Duration       Channel_Used  \
0  NY-CMP-1000  Social Media  College Students        21  WhatsApp, YouTube   

   Impressions  Clicks  Leads  Conversions  Revenue  Acquisition_Cost   ROI  \
0        57804    6156   3616         2355  1867515            111.03  6.14   

  Language  Engagement_Score  Customer_Segment        Date  
0    Hindi             20.98  College Students  29-04-2025  

----------------------------------------
Brand : Purplle
Shape : (55555, 16)
Columns : ['Campaign_ID', 'Campaign_Type', 'Target_Audience', 'Duration', 'Channel_Used', 'Impressions', 'Clicks', 'Leads',

➡️ All three dataset have 55555 rows and 16 columns with no missing values and no duplicates. They seem to have the exact same columns and we can confirm it.

In [ ]:
# Checking if columns are identical across all 3 files
assert df_nykaa.columns.tolist() == df_purplle.columns.tolist() == df_tira.columns.tolist(), \
    "Column mismatch between files !"
print("All columns are consistent across files.")

All columns are consistent across files.


➡️ Since all 3 files look correct and consistent, we can proceed to the merge.

In [ ]:
# Adding a 'Brand' column to each dataset
df_nykaa["Brand"] = "Nykaa"
df_purplle["Brand"] = "Purplle"
df_tira["Brand"] = "Tira Beauty"

# Stacking them vertically
df = pd.concat([df_nykaa, df_purplle, df_tira], ignore_index=True)

# Verifying merge integrity
print("Expected rows :", len(df_nykaa) + len(df_purplle) + len(df_tira))
print("Actual rows after merge :", len(df))
print("Match :", len(df) == len(df_nykaa) + len(df_purplle) + len(df_tira))

# Confirming all 3 brands are present
print("\nBrands in dataset :\n", df["Brand"].value_counts())

df.head(5)

Expected rows : 166665
Actual rows after merge : 166665
Match : True

Brands in dataset :
 Brand
Nykaa          55555
Purplle        55555
Tira Beauty    55555
Name: count, dtype: int64


,Campaign_ID,Campaign_Type,Target_Audience,Duration,Channel_Used,Impressions,Clicks,Leads,Conversions,Revenue,Acquisition_Cost,ROI,Language,Engagement_Score,Customer_Segment,Date,Brand
0,NY-CMP-1000,Social Media,College Students,21,"WhatsApp, YouTube",57804,6156,3616,2355,1867515,111.03,6.14,Hindi,20.98,College Students,29-04-2025,Nykaa
1,NY-CMP-1001,Paid Ads,Tier 2 City Customers,18,YouTube,91801,3321,1971,1357,1046247,180.83,3.26,Hindi,7.24,College Students,06-04-2025,Nykaa
2,NY-CMP-1002,Influencer,Youth,23,"WhatsApp, Google, YouTube",15536,2182,952,755,197055,90.60,1.88,English,25.03,College Students,14-01-2025,Nykaa
3,NY-CMP-1003,Email,Working Women,18,"YouTube, Facebook, Instagram",88114,8413,2231,947,376906,249.07,0.60,Hindi,13.15,College Students,04-06-2025,Nykaa
4,NY-CMP-1004,Paid Ads,College Students,10,"Facebook, Instagram",96871,3743,2060,1258,518296,228.60,0.80,Hindi,7.29,Tier 2 City Customers,29-12-2024,Nykaa


➡️ This confirms the merge was successful and no rows were lost or duplicated during the concat.
We can now verify the data types.

In [ ]:
# Standardizing column names to lowercase with underscores for consistency
df.columns = df.columns.str.lower().str.replace(" ", "_")
print("Renamed columns :\n", df.columns.tolist())

Renamed columns :
 ['campaign_id', 'campaign_type', 'target_audience', 'duration', 'channel_used', 'impressions', 'clicks', 'leads', 'conversions', 'revenue', 'acquisition_cost', 'roi', 'language', 'engagement_score', 'customer_segment', 'date', 'brand']


In [ ]:
# Inspecting data types to identify what actually needs changing
print (df.dtypes)

campaign_id          object
campaign_type        object
target_audience      object
duration              int64
channel_used         object
impressions           int64
clicks                int64
leads                 int64
conversions           int64
revenue               int64
acquisition_cost    float64
roi                 float64
language             object
engagement_score    float64
customer_segment     object
date                 object
brand                object
dtype: object


➡️ Let's convert:

- date : object → datetime to enable time-based analysis
- revenue : int → float to allow decimal precision in calculations

In [ ]:
# Fixing data types to ensure accurate analysis

# Date
df["date"] = pd.to_datetime(df["date"], format="%d-%m-%Y")

# Revenue
df["revenue"] = df["revenue"].astype("float64")

print("Updated data types :\n", df.dtypes)

# Confirm type conversion didn't introduce unexpected nulls or invalid dates
print(f"\nNulls after type fixing :{df[["revenue", "date"]].isnull().sum()}")
print(f"\nDate range : {df['date'].min().date()} to {df['date'].max().date()}")

Updated data types :
 campaign_id                 object
campaign_type               object
target_audience             object
duration                     int64
channel_used                object
impressions                  int64
clicks                       int64
leads                        int64
conversions                  int64
revenue                    float64
acquisition_cost           float64
roi                        float64
language                    object
engagement_score           float64
customer_segment            object
date                datetime64[ns]
brand                       object
dtype: object

Nulls after type fixing :revenue    0
date       0
dtype: int64

Date range : 2024-07-01 to 2025-06-24


➡️ The dataset covers ~12 months : July 2024 to June 2025. That will enable analysis such as seasonality or semester-over-semester comparisons.

📌 Our dataset is now clean and ready for quality analysis.

#### **2- Data Quality Check**

Before analyzing the data, let's assess its quality and reliability.
This section covers missing values, duplicates, statistical summary and outlier detection.

In [ ]:
# Verifying campaign_id is a unique identifier across the merged dataset
total_rows = len(df)
unique_ids = df["campaign_id"].nunique()

print(f"Total rows : {total_rows}")
print(f"Unique campaign_ids : {unique_ids}")
print(f"campaign_id is unique : {total_rows == unique_ids}")

Total rows : 166665
Unique campaign_ids : 166665
campaign_id is unique : True


➡️ Missing values and duplicates were already verified at the individual file level in Section 1 and none were found across all 3 datasets.

Since *campaign_id* was confirmed as unique identifier, cross-file duplicates is structurally impossible.

We can directly proceed to statistical summary and outlier detection.


In [ ]:
# Statistical Summary
# High-level overview of numeric columns
# Looking for suspicious min/max values, unexpected zeros or unrealistic ranges

df.describe().round(2)

,duration,impressions,clicks,leads,conversions,revenue,acquisition_cost,roi,engagement_score,date
count,166665.00,166665.00,166665.00,166665.00,166665.00,166665.00,166665.00,166665.00,166665.00,166665
mean,17.49,55060.85,4682.37,1871.51,1029.09,513906.62,376.09,2.69,13.77,2024-12-20 16:12:39.463594496
min,5.00,10001.00,202.00,48.00,17.00,3895.00,8.18,-0.99,2.56,2024-07-01 00:00:00
25%,11.00,32566.00,2109.00,779.00,401.00,177661.00,106.73,0.04,8.38,2024-09-25 00:00:00
50%,17.00,55110.00,3904.00,1476.00,776.00,359250.00,208.51,1.23,13.59,2024-12-21 00:00:00
75%,24.00,77574.00,6688.00,2598.00,1404.00,684797.00,427.57,3.58,18.79,2025-03-17 00:00:00
max,30.00,100000.00,14944.00,8876.00,6686.00,4579910.00,15473.16,79.30,30.99,2025-06-24 00:00:00
std,7.50,25970.65,3176.50,1429.59,858.01,487644.25,534.94,4.48,6.33,NaN


In [ ]:
# --- 2.4 Outlier Detection ---
# Using boxplots to visually identify outliers in key numeric columns

numeric_cols = ["impressions", "clicks", "leads", "conversions",
                "revenue", "acquisition_cost", "roi", "engagement_score"]

fig, axes = plt.subplots(2, 4, figsize=(18, 8))
axes = axes.flatten()

for i, col in enumerate(numeric_cols):
    axes[i].boxplot(df[col].dropna())
    axes[i].set_title(col)
    axes[i].set_xlabel("")

plt.suptitle("Outlier Detection — Key Numeric Columns", fontsize=14, y=1.02)
plt.tight_layout()
plt.show()

#### **3- Univariate Analysis**

#### **4- Bivariate Analysis**

#### **5- 360° Brand Analysis**

#### **6- Key Findings & Recommendations**